### ISSUE TITLE
Use WASM Websocket when available

### ISSUE THREAD

**Issue by Daddoon** [Daddoon/BlazorMobile#77]
We can't actually migrate to WASM Websocket.
We need the following issues to be done:

https://github.com/aspnet/AspNetCore/issues/10489

and

https://github.com/mono/mono/issues/9567 (already done)

**Comment by Monarch73** [comment]
10489 is done now .. :)

**Comment by Daddoon** [comment]
Ho great ! I will try to do some update about this on next release !

**Comment by Daddoon** [comment]
Not enough informations still for native support (at least from what i have found).
I'm asked some information.

As the project can still work "as is" without this evolution, i'm moving this task again.

**Comment by Daddoon** [comment]
I will try to make a test, as i have a feedback from Daniel Roth about Websocket usage.

As from the devblog here: https://devblogs.microsoft.com/aspnet/blazor-webassembly-3-2-0-preview-1-release-now-available/#comment-1385

**Question:**

```
Sorry if this is duplicate, but i didn’t find any answer to this question here, and even by reading aspnet issues on GitHub, nor the documentation:

Does the new Microsoft.AspNetCore.SignalR.Client package support connections to a regular WebSocket server (not managed by a SignalR server hub) ?
I knew there was some limitations some times ago about using Websockets API on Blazor WASM.

Does Microsoft.AspNetCore.SignalR.Client issue this requirement ? Otherwise, what is the alternative ? Does this kind of scenario is still impossible from C# in Blazor WASM ?

I would like to refactor any C# to Javascript code on my project for Websocket usage.

Regards,
Guillaume
```

**Answer:**

```
Hi Guillaume,

The .NET SignalR client is specific to SignalR scenarios. If you just want to connect over a WebSocket you should now be able to do that using the normal ClientWebSocket API in .NET: https://docs.microsoft.com/dotnet/api/system.net.websockets.clientwebsocket. Let us know if you run into any issues with that.

I hope this helps!

Daniel Roth
```

**Comment by Daddoon** [comment]
Nearly OK, but blocked on the **ClientWebSocket** WASM implementation that does not behave correctly.

It would work on .NET Core / Server-Side version. Waiting for feedback or a fix from Microsoft.

I have opened an issue here: https://github.com/dotnet/aspnetcore/issues/19468

**Comment by Daddoon** [comment]
Moving this task to "Task Ideas" until i have additionnals informations from Microsoft.

**Comment by Daddoon** [comment]
For tracking purpose: https://github.com/mono/mono/issues/19100
